core/src/main/java/org/elasticsearch/common/rounding/TimeZoneRounding.java

```diff
@@ -64,7 +64,9 @@ public Builder(TimeValue interval) {
         }
 
         public Builder timeZone(DateTimeZone timeZone) {
-            this.timeZone = timeZone;
+            if (timeZone != null) {
```

Comment:
maybe we should check that the timeZone argument is not null and fix callers to not call this method with a null timezone?